src/go/k8s/pkg/resources/configuration/configuration_modes.go

```diff
@@ -0,0 +1,192 @@
+package configuration
+
+import (
+	"reflect"
+	"strconv"
+	"strings"
+
+	"github.com/redpanda-data/redpanda/src/go/rpk/pkg/config"
+	"github.com/spf13/afero"
+)
+
+// GlobalConfigurationMode changes the behavior of the global configuration when reading/writing properties.
+type GlobalConfigurationMode interface {
+	SetAdditionalRedpandaProperty(targetConfig *GlobalConfiguration, key string, value interface{})
+	GetAdditionalRedpandaProperty(targetConfig *GlobalConfiguration, key string) interface{}
+	SetAdditionalFlatProperties(targetConfig *GlobalConfiguration, props map[string]string) error
+}
+
+var (
+	// GlobalConfigurationModeClassic puts everything in "redpanda.yaml"
+	GlobalConfigurationModeClassic GlobalConfigurationMode = globalConfigurationModeClassic{}
+	// GlobalConfigurationModeCentralized splits properties between "redpanda.yaml" and ".bootstrap.yaml"
+	GlobalConfigurationModeCentralized GlobalConfigurationMode = globalConfigurationModeCentralized{}
+	// GlobalConfigurationModeMixed puts central configuration properties in both "redpanda.yaml" and ".bootstrap.yaml"
```

Comment:
Node/local configuration goes in `redpanda.yaml`, and it can contain also central configuration properties. While `.bootstrap.yaml` can only contain central configuration properties.

The format is also different. E.g.

`redpanda.yaml` has central properties nested under `redpanda`:

```yaml
# ...
redpanda:
  cloud_storage_region: us-west-1
```

While `.bootstrap.yaml` is k/v:

```
cloud_storage_region: us-west-1
```

Properties like `redpanda.cloud_storage_cache_directory` can only go in `redpanda.yaml`.